src/config/config.ts

```diff
@@ -550,7 +552,8 @@ export class Config implements IConfig {
 
   protected _shell(): string {
     let shellPath
-    const {SHELL, COMSPEC} = process.env
+    const COMSPEC = process.env.COMSPEC
+    const SHELL = process.env.SHELL ?? osUserInfo().shell?.split(sep)?.pop()
     if (SHELL) {
       shellPath = SHELL.split('/')
```

Comment:
this conditional will be executed on Windows too. if `process.env.SHELL` is `undefined` then node throw a `Cannot read properties of null` err since `userInfo().shell` is always `null`.

Reply:
```suggestion
      shellPath = SHELL?.split('/')
```

Reply:
@cristiand391 line 557 guards the SHELL.split(), can never get here if undefined.